README.md

```diff
@@ -1,90 +1,90 @@
-    + [3.23 examples/sv1-client-and-server](#323-examplessv1-client-and-server)
-    + [3.24 examples/sv2-proxy](#324-examplessv2-proxy)
-    + [3.25 examples/template-provider-test](#325-examplestemplate-provider-test)
-    + [3.26 test](#326-test)
-    + [3.27 experimental](#327-experimental)
-  * [4. Branches](#4-branches)
+  - [1. Goals](#1-goals)
+  - [2. Common Use Cases](#2-common-use-cases)
+    - [2.1 Miners](#21-miners)
+    - [2.2 Pools](#22-pools)
+  - [3. Structure](#3-structure)
+    - [3.01 protocols](#301-protocols)
+    - [3.02 protocols/v1](#302-protocolsv1)
+    - [3.03 protocols/v2](#303-protocolsv2)
+    - [3.04 protocols/v2/const-sv2](#304-protocolsv2const-sv2)
+    - [3.05 protocols/v2/binary-sv2](#305-protocolsv2binary-sv2)
+    - [3.06 protocols/v2/framing-sv2](#306-protocolsv2framing-sv2)
+    - [3.07 protocols/v2/codec-sv2](#307-protocolsv2codec-sv2)
+    - [3.08 protocols/v2/subprotocols](#308-protocolsv2subprotocols)
+    - [3.09 protocols/v2/sv2-ffi](#309-protocolsv2sv2-ffi)
+    - [3.10 protocols/v2/noise-sv2](#310-protocolsv2noise-sv2)
+    - [3.11 protocols/v2/roles-logic-sv2](#311-protocolsv2roles-logic-sv2)
+    - [3.12 utils/buffer](#312-utilsbuffer)
+    - [3.13 utils/network-helpers](#313-utilsnetwork-helpers)
+    - [3.14 roles/mining-proxy](#314-rolesmining-proxy)
+    - [3.15 roles/pool](#315-rolespool)
+    - [3.16 roles/test-utils/pool](#316-rolestest-utilspool)
+    - [3.17 roles/test-utils/mining-device](#317-rolestest-utilsmining-device)
+    - [3.18 examples](#318-examples)
+    - [3.19 examples/interop-cpp](#319-examplesinterop-cpp)
+    - [3.20 examples/interop-cpp-no-cargo](#320-examplesinterop-cpp-no-cargo)
+    - [3.21 examples/ping-pong-with-noise](#321-examplesping-pong-with-noise)
+    - [3.22 examples/ping-pong-without-noise](#322-examplesping-pong-without-noise)
+    - [3.23 examples/sv1-client-and-server](#323-examplessv1-client-and-server)
+    - [3.24 examples/sv2-proxy](#324-examplessv2-proxy)
+    - [3.25 examples/template-provider-test](#325-examplestemplate-provider-test)
+    - [3.26 test](#326-test)
+    - [3.27 experimental](#327-experimental)
+  - [4. Branches](#4-branches)
 
 ## 1. Goals
-The goals of this project is to provide:
 
-1. A robust set of Stratum V2 (Sv2) primitives as Rust library crates that can be used by anyone
-   that wants to expand the protocol or implement a role (e.g. a pool wanting to support Sv2, a
-   mining-device/hashrate producer that wants to integrate Sv2 into their firmware, a Bitcoin node
-   that wants to implement Template Provider capabilities to build the `blocktemplate`).
+The goals of this project are to provide:
 
-1. The above Rust primitives as a C library, making it available to non-Rust users via a C
-   bindings.
+1. A robust set of Stratum V2 (Sv2) primitives as Rust library crates which anyone can use
+   to expand the protocol or implement a role. For example:
+   a) Pools supporting Sv2
```

Comment:
When creating these sub-bullet points - `a) , b), c)` don't appear on new lines when the markdown is rendered

Reply:
Fixed it, thanks!